### PR TITLE
Make Modifiers not extend from Positioned

### PR DESCRIPTION
**Pull request by odersky**
Modifiers was made a subclass of Positioned since it contains elements that
are Positioned: source modifiers and annotations. That way, we got the navigation
logic to get to the elements for free. However, a lot of modifiers are synthetic,
and it seems artificial to give these a position. With the addition of a source
component to position the balance swings the other way. It now seems preferable
to have Modifiers not extend Positions and to consider Modifier as a as a separate
case for navigation instead.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/parsing/Parsers.scala [hunk 1]
```diff
@@ -2016,7 +2013,7 @@ object Parsers {
             if (in.token == EQUALS) { in.nextToken(); expr() }
             else EmptyTree
           if (implicitOffset >= 0) {
-            mods = mods.withPos(mods.pos.union(Position(implicitOffset, implicitOffset)))
+            //mods = mods.withPos(mods.pos.union(Position(implicitOffset, implicitOffset)))
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/parsing/Parsers.scala]: This should be removed

File: compiler/src/dotty/tools/dotc/parsing/Parsers.scala [hunk 2]
```diff
@@ -2549,8 +2545,10 @@ object Parsers {
         setLastStatOffset()
         if (in.token == PACKAGE) {
           val start = in.skipToken()
-          if (in.token == OBJECT)
-            stats += objectDef(start, atPos(start, in.skipToken()) { Modifiers(Package) })
+          if (in.token == OBJECT) {
+            in.nextToken()
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/parsing/Parsers.scala]: Why not `skipToken`?

File: compiler/src/dotty/tools/dotc/parsing/Parsers.scala [hunk 3]
```diff
@@ -2710,7 +2708,8 @@ object Parsers {
         if (in.token == PACKAGE) {
           in.nextToken()
           if (in.token == OBJECT) {
-            ts += objectDef(start, atPos(start, in.skipToken()) { Modifiers(Package) })
+            in.nextToken()
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/parsing/Parsers.scala]: Why not `skipToken`?

Comment (odersky) [reply compiler/src/dotty/tools/dotc/parsing/Parsers.scala]: We don't need the old position returned by skipToken()

**Comment by odersky** [comment]
This is part of #5664, where I also addressed review comments.

**Pull request closed without merging**